modules/foreman_role.py

```diff
@@ -0,0 +1,134 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# (c) 2019 Christoffer Reijer (Basalt AB)
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+DOCUMENTATION = '''
+---
+module: foreman_role
+short_description: Manage Foreman Roles using Foreman API
+description:
+  - Create and Delete Foreman Roles using Foreman API
+version_added: "2.9"
+author:
+  - "Christoffer Reijer (@ephracis) Basalt AB"
+requirements:
+  - "apypie"
+options:
+  name:
+    description: The name of the role
+    required: true
+  description:
+    description: Description of the role
+    required: false
+  locations:
+    description: List of locations the role should be assigned to
+    required: false
+    default: None
+    type: list
+  organizations:
+    description: List of organizations the role should be assigned to
+    required: false
+    default: None
+    type: list
+  server_url:
+    description: foreman url
+    required: true
+  username:
+    description: foreman username
+    required: true
+  password:
+    description: foreman user password
+    required: true
+  validate_certs:
+    aliases: [ verify_ssl ]
+    description: verify ssl connection when communicating with foreman
+    default: true
+    type: bool
+  state:
+    description: domain presence
+    default: present
+    choices: ["present", "absent"]
+'''
+
+EXAMPLES = '''
+- name: role
+  foreman_role:
+    name: "test"
+    description: "test role"
```

Comment:
Can we use more meaningful data here? Something like "Provisioner" "Only provision on libvirt" or something like that

Reply:
Absolutely. Fixed!